Userland/Libraries/LibCore/System.cpp

```diff
@@ -308,4 +308,17 @@ ErrorOr<void> chown(StringView pathname, uid_t uid, gid_t gid)
 #endif
 }
 
+ErrorOr<struct passwd> getpwnam(StringView name)
+{
+    ::setpwent();
+    while (auto* pw = ::getpwent()) {
+        if (pw->pw_name == name)
+            return *pw;
+    }
+    if (errno == 0)
```

Comment:
Hmm, I'm not sure we can rely on `errno` being zero when there's no error.
My understanding is that POSIX functions are only expected to update `errno` when also returning a value that indicates an error occurred.

Reply:
Looking at the man page for `setpwent()` and `getpwent()`: 
> If an error occurs, errno is set appropriately.  If one wants to check errno after the  call,      it should be set to zero before the call.

Amending the function to set `errno` to zero in the beginning of the call and then checking at the end of the call seems appropriate from my understanding.

Reply:
Since I do not see us set `errno = 0` any other places at the beginning of calls I for now just updated it to check for `errno != 0` after `setpwent()` and then return `Error::from_syscall` if true.

Reply:
Scracth my comments above.

I think this should resolve errors from `setpwent` and `getpwent` in `getpwnam()` by checking `errno` after each call. Same type of implementation also in `getgrnam` https://github.com/SerenityOS/serenity/pull/11101/commits/c4d2bd2fc63bb3172b1e332139c2ed1d780a6244